### PR TITLE
NK-103 Camera rotation now working when parented

### PR DESCRIPTION
**Pull request by antopilo**
FIXED BUG 🐛🐛🐛🪓

### CODE REVIEW

File: Editor/src/Misc/GizmoDrawer.cpp [hunk 1]
```diff
@@ -259,7 +259,7 @@ void GizmoDrawer::DrawGizmos(Ref<Scene> scene)
 		flatShader->SetUniformVec4("u_Color", Vector4(0.0f, 1.0f, 0.4f, 1.0f));
 
 		const auto scaledTransform = glm::scale(transformComponent.GetGlobalTransform(), Vector3(0.25f, 0.25f, 0.25f));
-		renderList.AddToRenderList(_gizmos["player"]->GetMeshes()[0], scaledTransform);
+		//renderList.AddToRenderList(_gizmos["player"]->GetMeshes()[0], scaledTransform);
```
Comment (emerycp) [inline Editor/src/Misc/GizmoDrawer.cpp]: do we keep this comment?

File: Nuake/src/Core/Maths.cpp [hunk 2]
```diff
@@ -2,12 +2,80 @@
...
+    q.y = cr * sp * cy + sr * cp * sy;
+    q.z = cr * cp * sy - sr * sp * cy;
+    return q;
 }
 
 Vector3 Nuake::QuatToDirection(const Quat& quat)
 {
-	return glm::rotate(glm::inverse(quat), glm::vec3(1.0, 0.0, 0.0));
+	return glm::normalize(quat * Vector3(0, 0, -1));
+	//return glm::normalize(glm::rotate(glm::inverse(quat), glm::vec3(-1.0, 0.0, 0.0)));
```
Comment (emerycp) [inline Nuake/src/Core/Maths.cpp]: comments?

File: Nuake/src/Rendering/Camera.cpp [hunk 3]
```diff
@@ -52,7 +52,23 @@ namespace Nuake
...
+		//cam->cameraDirection.x = cos(glm::radians(Yaw)) * cos(glm::radians(Pitch));
+		//cam->cameraDirection.y = sin(glm::radians(Pitch));
+		//cam->cameraDirection.z = sin(glm::radians(Yaw)) * cos(glm::radians(Pitch));
+		//cam->cameraFront = glm::normalize(cam->cameraDirection);
+		//cam->cameraRight = glm::normalize(glm::cross(cam->up, cam->cameraFront));
+
+		//glm::quatLookAt();
+		//Direction = glm::normalize(direction);
+		//Right = glm::normalize(glm::cross(Vector3(0, 1, 0), Direction));
+		//Up = glm::normalize(glm::cross(Direction, Right));
```
Comment (emerycp) [inline Nuake/src/Rendering/Camera.cpp]: I guess we're keeping this for when we'll do the feature

File: Nuake/src/Rendering/Camera.cpp [hunk 4]
```diff
@@ -52,7 +52,23 @@ namespace Nuake
 		//cam->cameraRight = glm::normalize(glm::cross(cam->up, cam->cameraFront));
 		Direction = glm::normalize(direction);
 		Right = glm::normalize(glm::cross(Vector3(0, 1, 0), Direction));
-		//Up = glm::normalize(glm::cross(Direction, Right))
+		m_View = lookAt(Translation, Translation + glm::normalize(Direction), Vector3(0, 1, 0));
+
+		//Up = glm::normalize(glm::cross(Direction, Right));
```
Comment (emerycp) [inline Nuake/src/Rendering/Camera.cpp]: comments?

File: Nuake/src/Rendering/Camera.cpp [hunk 5]
```diff
@@ -70,17 +86,24 @@ namespace Nuake
 		return m_Perspective;
 	}
 
+	void Camera::SetTransform(const Matrix4& transform)
+	{
+		m_View = transform;
+
```
Comment (emerycp) [inline Nuake/src/Rendering/Camera.cpp]: remove empty line
```suggestion
```

File: Nuake/src/Rendering/Camera.cpp [hunk 6]
```diff
@@ -70,17 +86,24 @@ namespace Nuake
...
+		m_View = transform;
+		
+	}
+
 	Matrix4 Camera::GetTransform()
 	{
-		glm::mat4 tr = lookAt(Translation, Translation + Direction, Vector3(0, 1, 0));
+		return m_View;
+		glm::mat4 tr = lookAt(Translation, Translation + glm::normalize(Direction), Vector3(0, 1, 0));
 		return tr;
```
Comment (emerycp) [inline Nuake/src/Rendering/Camera.cpp]: is this return statement reachable? there seems to have 2 of them in this method.

**Comment by emerycp** [comment]
homer-simpsons

**Review by emerycp — APPROVED** [review]
Good job. 
HandshakeOfTheGodsGIF

**Comment by antopilo** [comment]
Note: the last bug with this PR is that the camera gizmo is backwards, this can be fixed by updating the camera gizmo. 

Will eventually move towards 2D sprites for representing cameras, so this is a non-issue for now.

**Pull request merged**